src/test/scala/spark/dicom/deidentifier/TestDicomDeidentifier.scala

```diff
@@ -0,0 +1,202 @@
+    with WithSpark {
+  import TestDicomDeidentifier._
+
+  val logger = {
+    val logger = LogManager.getLogger(getClass.getName);
+    logger.setLevel(Level.DEBUG)
+    logger
+  }
+
+  override protected def afterAll(): Unit = {
+    spark.stop
+  }
+
+  val SOME_DROPPED_COL = keywordOf(Tag.ContainerComponentID)
+  val SOME_DUMMY_DATE_COL = keywordOf(Tag.ContentDate)
+  val SOME_DUMMY_TIME_COL = keywordOf(Tag.ContentTime)
+  val SOME_DUMMY_DATE_TIME_COL = keywordOf(Tag.AttributeModificationDateTime)
+  val SOME_DUMMY_STRING_COL = keywordOf(Tag.AnnotationGroupLabel)
+  val SOME_ZERO_INT_COL = keywordOf(Tag.PregnancyStatus)
+  val SOME_ZERO_STRING_COL = keywordOf(Tag.ConceptualVolumeDescription)
+  val SOME_EMPTY_STRING_COL = keywordOf(Tag.AccessionNumber)
+  val SOME_CLEANED_COL = keywordOf(Tag.AcquisitionComments)
+
+  describe("Spark") {
+    it("Deidentifies DICOM dataframe using basic profile") {
+      var df = spark.read
+        .format("dicomFile")
+        .load(SOME_DICOM_FILEPATH)
+
+      df = deidentify(df)
+
+      val row = df.first
+
+      assert(
+        row.getAs[String](SOME_DUMMY_DATE_COL)
+          === DUMMY_DATE
+      )
+      assert(
+        row.getAs[String](SOME_DUMMY_TIME_COL)
+          === DUMMY_TIME
+      )
+      assert(
+        row.getAs[String](SOME_DUMMY_DATE_TIME_COL)
+          === DUMMY_DATE_TIME
+      )
+      assert(
+        row.getAs[String](SOME_DUMMY_STRING_COL)
+          === DUMMY_STRING
+      )
+      assert(
+        row.getAs[String](SOME_EMPTY_STRING_COL)
+          === EMPTY_STRING
+      )
+      assertThrows[IllegalArgumentException] {
+        row.fieldIndex(SOME_DROPPED_COL)
+      }
+      // No zero string/int cols in basic profile. To be added later
+      // assert(
+      //   row.getAs[Int](SOME_ZERO_INT_COL)
+      //     === ZERO_INT
+      // )
+      // assert(
+      //   row.getAs[String](SOME_ZERO_STRING_COL)
+      //     === ZERO_STRING
+      // )
+    }
+    it("Deidentifies DICOM dataframe using an option") {
+      var df = spark.read
+        .format("dicomFile")
+        .load(SOME_DICOM_FILEPATH)
+
+      val options = Seq(
+        CleanDesc()
+      )
+
+      df = deidentify(df, options)
+
+      val row = df.first
+
+      assert(
+        row.getAs[String](SOME_CLEANED_COL)
+          === "ToClean"
+      )
+    }
+  }
+  describe("Deidentifier") {
+    it("getAction returns default action when no options given") {
+      val deidElem = DicomDeidElem(
+        tag = 0,
+        name = "test",
+        keyword = "test",
+        action = "X",
+        retainUidsAction = Some("D")
+      )
+      assert(DicomDeidentifier.getAction(deidElem) === Drop())
+    }
+    it("getAction returns correct action when option is given") {
+      val deidElem = DicomDeidElem(
+        tag = 0,
+        name = "test",
+        keyword = "test",
+        action = "X",
+        retainUidsAction = Some("D")
+      )
+      val options = Seq(RetainUids())
```

Comment:
I believe you can omit the parenthesis? Worth checking

 ```suggestion
      val options = Seq(RetainUids)
```

Reply:
only if `case object` I believe...